### PR TITLE
test: Fix incorrect rawkv case test_raw_put_key_guard

### LINKED ISSUE #13599: rawkv: unit test "test_storage::test_raw_put_key_guard" is not correct and unstable
## Bug Report

Unit test "test_storage::test_raw_put_key_guard" is not correct [here](https://github.com/tikv/tikv/blob/11a340c3ec54fdbe1582a4b6eb669f43aa924ba2/tests/failpoints/cases/test_storage.rs#L1524). `must_get_none(&cluster.get_engine(1), b"rk3");` is expected to get a value as `v3`.

Moreover, this case is unstable.
See [here](https://ci.pingcap.net/blue/organizations/jenkins/tikv_ghpr_test/detail/tikv_ghpr_test/27997/pipeline/)
```
[2022-10-13T09:32:30.271Z] failures:
[2022-10-13T09:32:30.271Z]     cases::test_storage::test_raw_put_key_guard
[2022-10-13T09:32:30.271Z] 
[2022-10-13T09:32:30.271Z] test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 225 filtered out; finished in 0.62s
[2022-10-13T09:32:30.271Z] 
[2022-10-13T09:32:30.271Z] 
[2022-10-13T09:32:30.271Z] --- TRY 3 STDERR:                              tests::failpoints cases::test_storage::test_raw_put_key_guard ---
[2022-10-13T09:32:30.271Z] thread 'cases::test_storage::test_raw_put_key_guard' panicked at 'called `Option::unwrap()` on a `None` value', tests/failpoints/cases/test_storage.rs:1519:23
[2022-10-13T09:32:30.271Z] stack backtrace:
[2022-10-13T09:32:30.271Z]    0: rust_begin_unwind
[2022-10-13T09:32:30.271Z]              at /rustc/0f4bcadb46006bc484dad85616b484f93879ca4e/library/std/src/panicking.rs:584:5
[2022-10-13T09:32:30.271Z]    1: core::panicking::panic_fmt
[2022-10-13T09:32:30.271Z]              at /rustc/0f4bcadb46006bc484dad85616b484f93879ca4e/library/core/src/panicking.rs:142:14
[2022-10-13T09:32:30.271Z]    2: core::panicking::panic
[2022-10-13T09:32:30.271Z]              at /rustc/0f4bcadb46006bc484dad85616b484f93879ca4e/library/core/src/panicking.rs:48:5
[2022-10-13T09:32:30.271Z]    3: core::option::Option<T>::unwrap
[2022-10-13T09:32:30.271Z]    4: failpoints::cases::test_storage::test_raw_put_key_guard
[2022-10-13T09:32:30.271Z]              at /home/jenkins/tikv-src/tests/failpoints/cases/test_storage.rs:1519:16
[2022-10-13T09:32:30.271Z]    5: failpoints::cases::test_storage::test_raw_put_key_guard::{{closure}}
[2022-10-13T09:32:30.271Z]              at /home/jenkins/tikv-src/tests/failpoints/cases/test_storage.rs:1484:1
[2022-10-13T09:32:30.271Z]    6: core::ops::function::FnOnce::call_once
[2022-10-13T09:32:30.271Z]              at /rustc/0f4bcadb46006bc484dad85616b484f93879ca4e/library/core/src/ops/function.rs:248:5
[2022-10-13T09:32:30.271Z]    7: test_util::runner::run_test_with_hook::{{closure}}::{{closure}}
[2022-10-13T09:32:30.271Z]    8: core::ops::function::FnOnce::call_once{{vtable.shim}}
[2022-10-13T09:32:30.271Z]    9: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
[2022-10-13T09:32:30.271Z]              at /rustc/0f4bcadb46006bc484dad85616b484f93879ca4e/library/alloc/src/boxed.rs:1935:9
[2022-10-13T09:32:30.271Z] note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
[2022-10-13T09:32:30.271Z] thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: RpcFailure(RpcStatus { code: 14-UNAVAILABLE, message: "Socket closed", details: [] })', tests/failpoints/cases/test_storage.rs:1513:42
[2022-10-13T09:32:30.271Z] stack backtrace:
[2022-10-13T09:32:30.271Z]    0: rust_begin_unwind
[2022-10-13T09:32:30.271Z]              at /rustc/0f4bcadb46006bc484dad85616b484f93879ca4e/library/std/src/panicking.rs:584:5
[2022-10-13T09:32:30.271Z]    1: core::panicking::panic_fmt
[2022-10-13T09:32:30.271Z]              at /rustc/0f4bcadb46006bc484dad85616b484f93879ca4e/library/core/src/panicking.rs:142:14
[2022-10-13T09:32:30.271Z]    2: core::result::unwrap_failed
[2022-10-13T09:32:30.271Z]              at /rustc/0f4bcadb46006bc484dad85616b484f93879ca4e/library/core/src/result.rs:1814:5
[2022-10-13T09:32:30.271Z]    3: core::result::Result<T,E>::unwrap
[2022-10-13T09:32:30.271Z]    4: failpoints::cases::test_storage::test_raw_put_key_guard::{{closure}}
[2022-10-13T09:32:30.271Z]              at /home/jenkins/tikv-src/tests/failpoints/cases/test_storage.rs:1513:17
[2022-10-13T09:32:30.271Z] note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
[2022-10-13T09:32:30.271Z] 
[2022-10-13T09:32:30.271Z] error: test run failed
```

### What version of TiKV are you using?

master (11a340c3ec54fdbe1582a4b6eb669f43aa924ba2)

### What operating system and CPU are you using?

CI.

### Steps to reproduce

Submit PR to trigger Ci.

### What did you expect?

Case "test_storage::test_raw_put_key_guard" succeed.

### What did happened?

Case "test_storage::test_raw_put_key_guard" failed.

### PR DESCRIPTION
**Pull request by haojinming**
Signed-off-by: haojinming <jinming.hao@pingcap.com>

### What is changed and how it works?

Issue Number: Close #13599 

What's Changed:

```commit-message
The logic in `test_raw_put_key_guard` is incorrect, fix it. 
```

### Check List 

Tests 

- Unit test

Side effects

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by pingyu — APPROVED** [review]
LGTM~

File: tests/failpoints/cases/test_rawkv.rs [hunk 1]
```diff
@@ -271,3 +275,45 @@ fn test_region_merge() {
...
+    let copy_test_key = test_key.clone();
+    let copy_test_value = test_value.clone();
+    let apply_wait_timeout = 2000; // ms, assume send request and apply can be finished in 2s.
+    fail::cfg(pause_write_fp, "pause").unwrap();
+    let handle = thread::spawn(move || {
+        must_raw_put(&client, ctx, copy_test_key, copy_test_value);
+    });
+    thread::sleep(Duration::from_millis(apply_wait_timeout));
+
+    // Before raw_put finish, min_ts should be the ts of "key guard" of the raw_put
```
Comment (pingyu) [inline tests/failpoints/cases/test_rawkv.rs:306]: Nit: use the same name as "key guard" or "lock guard" here and line no 315.

**Review by zeminzhou — APPROVED** [review]
LGTM～

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**